src/App.tsx

```diff
@@ -14,11 +14,193 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[],
+  isStarted: boolean,
+  isReversed: boolean,
+  isSorted: Sorting,
+  length: number,
+};
+
+enum Sorting {
+  Default,
+  Asc,
+  Length,
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: [...goodsFromServer],
+    isStarted: false,
+    isReversed: false,
+    isSorted: Sorting.Default,
+    length: 1,
+  };
+
+  handleToStart = () => {
+    this.setState({ isStarted: true });
+  };
+
+  handleToReset = () => {
+    this.setState({
+      goods: [...goodsFromServer],
+      isReversed: false,
+      isSorted: Sorting.Default,
+      length: 1,
+    });
+  };
+
+  sortByReverse = () => {
+    this.setState({ isReversed: true });
```

Comment:
now your `reverse` is a one-time use
enable it to reverse every time it's clicked - use 
```suggestion
    this.setState({ isReversed: !this.state.isReversed });
```